pylint/testutils/primer/primer_clone_external.py

```diff
@@ -0,0 +1,39 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+import os
+import shutil
+
+import git
+
+from pylint.testutils.primer.primer_external_packages import (
+    PRIMER_DIRECTORY,
+    PackageToLint,
+)
+
+
+def _lazy_git_clone(data: PackageToLint, target_directory: str) -> None:
+    """Clones a repository while checking if it hasn't already been cloned"""
+    if os.path.exists(target_directory):
+        # Get SHA1 hash of latest commit on remote branch
+        remote_sha1_commit = (
+            git.cmd.Git().ls_remote(data.url, data.branch).split("\t")[0]
+        )
+        # Get SHA1 hash of latest commit on locally downloaded branch
+        local_sha1_commit = git.Repo(target_directory).head.object.hexsha
+
+        if remote_sha1_commit != local_sha1_commit:
+            # Remove directory and all its files
+            shutil.rmtree(target_directory)
+            git.Repo.clone_from(
+                data.url,
+                target_directory,
+            )
```

Comment:
I think a ``git pull`` would be a lot more efficient.

Reply:
I opted to remove and clone to avoid any issues with merge conflicts etc. During testing I used the primer to download repositories and then changed the local code to find crashes. I think that won't be possible anymore if we pull. I'm fine with changing to pulling.

Reply:
02b10b483b30a41b1f85da1d339a729e95343112

I did not manage to do a proper stash and stash pop (https://stackoverflow.com/questions/28265396/gitpython-how-to-git-stash-changes-to-a-gitpython-repository) but it's possible to pull with local modification as long as they're no conflicts.